### PR TITLE
added max selection count to picker options

### PR DESCRIPTION
**Pull request by stephkyl**

### CODE REVIEW

File: pick/__init__.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class Picker(object):
     :param options_map_func: (optional) a mapping function to pass each option through before displaying
     """
 
-    def __init__(self, options, title=None, indicator='*', default_index=0, multiselect=False, multi_select=False, min_selection_count=0, options_map_func=None):
+    def __init__(self, options, title=None, indicator='*', default_index=0, multiselect=False, multi_select=False, min_selection_count=0, max_selection_count=0, options_map_func=None, pre_select_all=False):
```
Comment (wong2) [inline pick/__init__.py]: `max_selection_count` should default to None

File: pick/__init__.py [hunk 2]
```diff
@@ -31,14 +31,21 @@ def __init__(self, options, title=None, indicator='*', default_index=0, multisel
         self.indicator = indicator
         self.multiselect = multiselect or multi_select
         self.min_selection_count = min_selection_count
+        self.max_selection_count = max_selection_count
         self.options_map_func = options_map_func
-        self.all_selected = []
+
+        if multiselect is False and pre_select_all is True:
+            raise ValueError('pre_select_all only works for multiple selections')
+        self.all_selected = list(range(len(options))) if pre_select_all else []
```
Comment (wong2) [inline pick/__init__.py]: `pre_select_all` doesn't seem to work well with `min_selection_count` and `max_selection_count`

Comment (stephkyl) [reply pick/__init__.py]: I did not mean to add this in the same commit, so I will remove it.

File: pick/__init__.py (hunk 1 above)
Comment (stephkyl) [reply pick/__init__.py]: Ok

**Pull request closed without merging**